util/merge_settings.py

```
@@ -0,0 +1,103 @@
+
+For the purposes of simplifying usage, non-existant files are considered
+equivalent to empty files, and the script will not crash.
+"""
+
+import os
+import sys
+import click
+import yaml
+
+
+def merge(settings_objects):
+    """
+    Merge multiple settings objects.
+
+    Arguments:
+        settings_objects (list of list of dict):
+            Each list element represents settings files, each sublist element
+            represents each yaml document in the settings file.  The settings
+            objects are given in order from lowest to highest priority.
+    """
+    merged_public = {}
+    merged_secrets = {}
+    for settings in settings_objects:
+        # yaml interprets voids as "null" and returns a python Null, but we
+        # really think of those as empty dicts when writing settings.
+        if settings[0]:
+            public = settings[0]
+        else:
+            public = {}
+        # also, yaml document which is completely absent should be considered
+        # empty dicts for the purposes of merging.
+        if len(settings) == 2 and settings[1]:
+            secrets = settings[1]
+        else:
+            secrets = {}
+
+        merged_public.update(public)
+        merged_secrets.update(secrets)
+
+    return (merged_public, merged_secrets)
+
+
+@click.command()
+@click.argument('settings_filenames', type=click.Path(), nargs=-1)
+def main(settings_filenames):
+    """
+    The only command, and main entry point for this script.
+
+    Arguments:
+        settings_filenames (tuple of unicode strings):
+            Paths to settings files (containing one or two yaml documents)
+            ordered from lowest to highest priority.
+    """
+    existing_settings_files = []
+    for filename in settings_filenames:
+        if os.path.exists(filename):
+            existing_settings_files.append(open(filename, 'r'))
+        else:
+            sys.stderr.write('skipping non-existant file: {}\n'.format(filename))
```

That's totally fair.  I had the settings files hard-coded in the recipe for the make target, and this made it possible to do that.  Will change to fail fast.